docs/Kube-API-examples.md

```diff
@@ -1,4 +1,48 @@
-#Assisted Installer Kube API CR examples
+# Hive Integration
+
+The goal of the Hive integration is to enable Assisted Installer capabilities on-premise in users' "Hub" clusters by installing clusters via Multi-cluster management, such as through [Hive](https://github.com/openshift/hive/) and [RHACM](https://github.com/open-cluster-management) (Red Hat Advanced Cluster Management).
+
+A full description of the enhancement is available [here](https://github.com/openshift/enhancements/blob/master/enhancements/installer/agent-based-installation-in-hive.md).
+
+For this integration, the Assisted Installer APIs are available via [CRDs](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/)
+## CRD Types
+
+### [ClusterDeployment](https://github.com/openshift/hive/blob/master/apis/hive/v1/clusterdeployment_types.go)
+The ClusterDeployment CRD is defined in the Hive repository, where additional fields are added for Agent Based installation.
+In the ClusterDeployment, the user can specify requirements like Networking, number of Control Plane and Workers nodes and more.
+
+The installation will start automatically if the required number of hosts is available, the hosts are ready to be installed and the Agents are approved.
+
+### [InstallEnv](https://github.com/openshift/assisted-service/blob/master/internal/controller/api/v1alpha1/installenv_types.go)
+The InstallEnv CRD represents the configuration needed to create the discovery ISO.
+The user can specify proxy settings, ignition overrides and specify NMState labels.
+
+When the ISO is ready, an URL will be available in the CR.
+
+
+### [NMStateConfig](https://github.com/openshift/assisted-service/blob/master/internal/controller/api/v1alpha1/nmstate_config_types.go)
+The NMStateConfig contains network configuration that will applied on the hosts.
+
+
+### [Agent](https://github.com/openshift/assisted-service/blob/master/internal/controller/api/v1alpha1/agent_types.go)
+The Agent CRD represents a Host that boot from an ISO and registered to a cluster.
+It will be created by Assisted Service when a host registers.
+In the Agent, the user can specify the hostname, role, installation disk and more.
+Also, the host hardware inventory and statuses are available.
+
+Note that if the Agent is not Approved, it will not be part of the installation.
+
+Here how to approve an Agent:
+
+`kubectl -n assisted-installer patch agents.adi.io.my.domain 120af504-d88e-46bd-bec2-b8b261db3b01 -p '{"spec":{"approved":true}}' --type merge`
+
+## Day 2 worker
```

Comment:
Do we have a day2 doc to reference here?

Reply:
I am not aware of any.